Update creating workspaces and managing variables docs (#2270)
hashicorp/terraform-website: content/cloud-docs/workspaces/creating.mdx

  -> **Note:** Depending on your provider, you may need to change which account's repositories are shown. If the repository is not listed, scroll to the bottom of the list and enter its ID in the text field.

1. Enter a **Workspace Name**. This defaults to the repository name, if applicable. The name must be unique within the organization and can include letters, numbers, dashes (`-`), and underscores (`_`). Refer to our [workspace naming recommendations](/cloud-docs/workspaces/naming).
1. Add an optional **Description** that will appear at the top of the workspace in the Terraform Cloud UI.

1. (Version control only) Open **Advanced options** to optionally configure the following additional settings:
   - [Terraform Working Directory][workdir]
   - [Automatic Run Triggering][trigger]
   - [VCS branch][branch]
   - [Include submodules on clone][submodules]

1. Click **Create workspace**.

For CLI and API-driven workspaces, the workspace overview page appears. For version control workspaces, the **Configure Terraform variables** page appears.

### Configure Terraform Variables (Version Control Only)

> After you create a new workspace from a version control repository, Terraform Cloud scans its configuration files for undefined [Terraform variables](/cloud-docs/workspaces/variables#terraform-variables) without default values and automatically creates those variables within the workspace. Terraform cannot perform successful runs until you set values for these undefined variables.

Reviewer: I not sure about the use of "undefined" here, but need to ponder what it would be instead. Did you find undefined used else where? Maybe Terraform variables with undefined values (i.e. it does not have a default value or value set in the terraform config).

Author: Laura came up with this terminology 😄 What makes you unsure about using "undefined" here?

Reviewer 2: good flag! I think we can probably just say "without default values" and remove "undefined". If it doesn't feel right as someone who works with the product, I respect that!

Reviewer: I  can see reading undefined as not present in the terraform config -- but i could be over reading this too

Reviewer 2: should we say "After you create a new workspace from a version control repository, Terraform Cloud scans its configuration files for [Terraform variables](/cloud-docs/workspaces/variables#terraform-variables) without default values and automatically creates those variables within the workspace. Terraform cannot perform successful runs until you set default values for these variables."

@ker-an @rberecka

Author: Will update!

Author: For the last sentence, we can probably remove the "default":

"Terraform cannot perform successful runs until you set values for these variables."

But aside from that, it makes sense! @rberecka can you have a look? You know more about the variables then I do 🙈

Reviewer: @laurapacilio I would remove the word default from the last part of the last sentence and then yes 👍

Author: I will update!

> After you create a new workspace from a version control repository, Terraform Cloud scans its configuration files for undefined [Terraform variables](/cloud-docs/workspaces/variables#terraform-variables) without default values and automatically creates those variables within the workspace. Terraform cannot perform successful runs until you set values for these undefined variables.

Reviewer: There is also a nuance with working directories at the moment. Maybe at the end something like:

```suggestion
After you create a new workspace from a version control repository, Terraform Cloud scans its configuration files for undefined [Terraform variables](/cloud-docs/workspaces/variables#terraform-variables) without default values and automatically creates those variables within the workspace. Terraform cannot perform successful runs until you set values for these undefined variables.

This new variable detection feature currently only works if the terraform config is in the root (default) working directory. If you are using a different working directory, you can enter and configure the variables manually from the workspace overview.
```

Reviewer: To include what Laura said above

```suggestion
After you create a new workspace from a version control repository, Terraform Cloud scans its configuration files for [Terraform variables](https://github.com/cloud-docs/workspaces/variables#terraform-variables) without default values and automatically creates those variables within the workspace. Terraform cannot perform successful runs until you set values for these variables."

This new variable detection feature currently only works if the terraform config is in the root (default) working directory. If you are using a different working directory, you can enter and configure the variables from the workspace overview.
```

Author: This is really great to know! I wonder if we could add this as a note - as we do later on in the docs. Let me take a look at this 😄

Reviewer: ooh -- if there's a notes section that sounds good!

Choose one of the following actions:

- To skip this step, click **Go to workspace overview**. Terraform Cloud still adds any undefined Terraform variables without default values to the workspace. You can set values for these variables later from within the workspace.
- To configure variables, enter a value for each variable and click **Save variables**.

## After Creating a Workspace

If you have already configured all Terraform variables, we recommend [manually starting a run](/cloud-docs/run/ui#manually-starting-runs) to prepare VCS-driven workspaces. You may also want to do one or more of the following actions:

- [Upload configuration versions](/cloud-docs/workspaces/configurations#providing-configuration-versions): If you chose the API or CLI-Driven workflow, you must upload configuration versions for the workspace.

> - [Edit input or environment variables](/cloud-docs/workspaces/variables): Input variables define the parameters of a Terraform configuration, and shell environment variables store credentials and customize Terraform's behavior.

Reviewer: I think you can remove the bits in this item about input variables -- since those are handled above with line 63.

```suggestion
- [Edit environment variables](/cloud-docs/workspaces/variables): Shell environment variables store credentials and customize Terraform's behavior.
```

- [Edit additional workspace settings](/cloud-docs/workspaces/settings): This includes notifications, permissions, and run triggers to start runs automatically.
- [Learn more about running Terraform in your workspace](/cloud-docs/run): This includes how Terraform processes runs within the workspace, run modes, run states, etc.

### VCS Connection

> If you connected a VCS repository to the workspace, Terraform Cloud automatically registers a webhook with your VCS provider. A workspace with no runs will not accept new runs from a VCS webhook, so you must [manually start at least one run](/cloud-docs/run/ui#manually-starting-runs).

Reviewer: love changing this from queue to start.

Author: that's all Laura!! 🙌🏽

1. Add an optional **Description** that will appear at the top of the workspace in the Terraform Cloud UI.

1. (Version control only) Open **Advanced options** to optionally configure the following additional settings:
   - [Terraform Working Directory][workdir]
   - [Automatic Run Triggering][trigger]
   - [VCS branch][branch]
   - [Include submodules on clone][submodules]

1. Click **Create workspace**.

For CLI and API-driven workspaces, the workspace overview page appears. For version control workspaces, the **Configure Terraform variables** page appears.

### Configure Terraform Variables (Version Control Only)

After you create a new workspace from a version control repository, Terraform Cloud scans its configuration files for [Terraform variables](/cloud-docs/workspaces/variables#terraform-variables) without default values and automatically creates those variables within the workspace. Terraform cannot perform successful runs until you set values for these variables.

>   -> **Note:** Your Terraform configuration needs to be in the root (default) working directory for this to work. If you're using a different working directory, configure variables from within the workspace.

Author: How does this sound @laurapacilio and @rberecka?

This is related to the comment Rachael made here: https://github.com/hashicorp/terraform-website/pull/2270#discussion_r864279781

Reviewer: 👍

1. Choose a repository from the filterable list. The **Configure settings** screen appears.

   Some VCS providers limit the list's size. If a repository isn't listed, you can still choose it by name; scroll to the bottom of the list and enter its ID in the text field.

   -> **Note:** For some VCS providers, this list includes a drop-down menu for changing which account's repositories are shown. Other providers combine all available accounts into a single list.

1. Enter a name for the workspace. This defaults to the repository name, if applicable. The name must be unique within the organization, and can include letters, numbers, dashes (`-`), and underscores (`_`). See our [advice for useful workspace names](/cloud-docs/workspaces/naming).

1. Optionally, click the **Advanced options** link on the third screen to configure some additional version control settings. (These settings not shown if you chose **No VCS connection**.) For information about these settings, see:
   - [Terraform Working Directory][workdir]
   - [Automatic Run Triggering][trigger]
   - [VCS branch][branch]
   - [Include submodules on clone][submodules]

> 1. Click the **Create workspace** button to confirm creation. The **Configure Terraform variables** screen appears. You can set values for undefined [Terraform variables](/cloud-docs/workspaces/variables#terraform-variables) within the configuration or click the **Go to workspace overview** button to skip this step.

Author: Should we mention something about variables with defaults? Any variables with default values won't show up on this screen.

Reviewer 2: oo yes we should! let me push up a change for that now!
